build-support/src/main/kotlin/com/squareup/wire/buildsupport/WireBuildPlugin.kt

```diff
@@ -0,0 +1,382 @@
+import org.jetbrains.dokka.gradle.DokkaTaskPartial
+import org.jetbrains.kotlin.gradle.dsl.KotlinCompile
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.tasks.KotlinJvmCompile
+
+private lateinit var wireGroupId: String
+private lateinit var wireVersion: String
+
+// I cannot get the wireBuild extension to work in projects included via `includeBuild` within
+// `build-support`, so doing it here for now :sad:
+private val PROJECT_TO_PUBLISH = listOf(
+  "wire-bom",
+  "wire-compiler",
+  "wire-gradle-plugin",
+  "wire-grpc-client",
+  "wire-grpc-server",
+  "wire-grpc-server-generator",
+  "wire-gson-support",
+  "wire-java-generator",
+  "wire-kotlin-generator",
+  "wire-moshi-adapter",
+  "wire-reflector",
+  "wire-runtime",
+  "wire-runtime-swift",
+  "wire-schema",
+  "wire-schema-tests",
+  "wire-swift-generator",
+)
+
+@Suppress("unused") // Invoked reflectively by Gradle.
+class WireBuildPlugin : Plugin<Project> {
+  private lateinit var libs: LibrariesForLibs
+
+  override fun apply(target: Project) {
+    libs = target.extensions.getByName("libs") as LibrariesForLibs
+
+    wireGroupId = target.property("GROUP") as String
+    wireVersion = target.property("VERSION_NAME") as String
+
+    target.extensions.add(
+      WireBuildExtension::class.java,
+      "wireBuild",
+      WireBuildExtensionImpl(target),
+    )
+
+    target.configureCommonSpotless()
+    target.configureCommonTesting()
+    target.configureCommonAndroid()
+    target.configureCommonKotlin()
+    target.configureCommonDistribution()
+    target.configureCommonJarManifest()
+
+    if (target.name in PROJECT_TO_PUBLISH) {
+      target.extensions.getByType(WireBuildExtension::class.java).publishing()
+    }
+
+    if (target.name == "wire-gradle-plugin") {
+      target.publishPluginsIfRelease()
+    }
+  }
+
+  private fun Project.publishPluginsIfRelease() {
+    // Note that we create the task in all cases for it'll be executed on CI every time.
+    tasks.register("publishPluginToGradlePortalIfRelease") {
+      // Snapshots cannot be released to the Gradle portal. And we don't want to release internal
+      // square builds.
+      if (wireVersion.endsWith("-SNAPSHOT") || wireVersion.contains("square")) return@register
+      dependsOn(":wire-gradle-plugin:publishPlugins")
+    }
+  }
+
+  private fun Project.configureCommonSpotless() {
+    plugins.apply("com.diffplug.spotless")
+    val spotless = extensions.getByName("spotless") as SpotlessExtension
+    val licenseHeaderFile = rootProject.file("gradle/license-header.txt")
+    spotless.apply {
+      // The nested build-support Gradle project contains Java sources. Use our root project to
+      // target its sources rather than duplicating the Spotless setup in multiple places.
+      java {
+        if (path == ":") {
+          target("build-support/settings/src/**/*.java")
+        } else {
+          target("src/**/*.java")
```

Comment:
Looks like you need an exclude here for test fixtures that have nested gradle builds

Reply:
Is it what's creating the task dependency problems with spotless?

Reply:
Got it, I saw the CI failures